src/store/addressbooks.js

```diff
@@ -101,21 +101,23 @@ const mutations = {
 	 * @param {Object} state
 	 * @param {Object} data
 	 * @param {Object} data.addressbook the addressbook
+	 * @param {String} data.sharee the sharee
+	 * @param {Boolean} data.group group
 	 */
-	shareAddressbook(state, addressbook, sharee) {
-		addressbook = state.addressbooks.find(search => search === addressbook)
+	shareAddressbook(state, { addressbook, sharee, group }) {
+		addressbook = state.addressbooks.find(search => search.id === addressbook.id)
 		let newSharee = {}
-		sharee.displayname = sharee
-		sharee.writable = false
-		addressbook.shares.append(newSharee)
+		newSharee.displayname = sharee
```

Comment:
You can do that directly in the object declaration

Reply:
💯